providers/fips/fipsprov.c

```diff
@@ -504,20 +506,29 @@ static const OSSL_ALGORITHM *fips_query(void *provctx, int operation_id,
 }
 
 void *ossl_c_locale() {
+# if !defined (OPENSSL_SYS_TANDEM)
     return (void *)loc;
+# else
+    return NULL;
+# endif
 }
 
 static int fips_init_casecmp(void) {
-# ifdef OPENSSL_SYS_WINDOWS
+# if defined (OPENSSL_SYS_WINDOWS)
     loc = _create_locale(LC_COLLATE, "C");
+    return (loc == (locale_t) 0) ? 0 : 1;
+# elif defined (OPENSSL_SYS_TANDEM)
+    return 1;
 # else
     loc = newlocale(LC_COLLATE_MASK, "C", (locale_t) 0);
-# endif
     return (loc == (locale_t) 0) ? 0 : 1;
+# endif
 }
 
 static void fips_deinit_casecmp(void) {
+# if !defined (OPENSSL_SYS_TANDEM)
```

Comment:
```suggestion
# if !defined (OPENSSL_USE_LOCALE)
```

Reply:
Replaced with OPENSSL_SUPPRESS_LOCALE_T supplied in `Configuration/50-nonstop.conf` so that other platforms can use it without needing to enumerate those in code.

Reply:
I do not think this is the right approach.